docs/source/extracting-specification.py

```diff
@@ -0,0 +1,60 @@
+# Requires python 3
+#
+# Reads the stored specification from a freshly created NWB file and exports it
+# into pretty printed JSON files.
+
+from datetime import datetime
+from dateutil.tz import tzlocal
+
+import os
+import io
+import json
+import sys
+
+import h5py
+
+from pynwb import NWBFile, NWBHDF5IO
+
+
+def createNWBFile():
+
+    nwbfile = NWBFile('my first synthetic recording',
+                      'EXAMPLE_ID',
+                      datetime.now(tzlocal()))
+
+    filename = 'specification-extraction.nwb'
+
+    with NWBHDF5IO(filename, 'w') as nwbio:
+        nwbio.write(nwbfile, cache_spec=True)
+
+    return filename
+
+
+def exportSpecification(filename):
+
+    with h5py.File(filename, 'r') as nwb:
+        spec = nwb[nwb.attrs['.specloc']]
+        for ns in spec.values():
+            for version in ns.values():
+                for sub in version.values():
+                    contents = json.loads(sub[()])
+
+                    def getName(o):
+                        return os.path.basename(o.name)
+
+                    filename = "{}_{}_{}_{}.json".format(getName(spec), getName(ns),
```

Comment:
I think it would be good to use the SpecWrite from HDMF to do the write:

https://github.com/hdmf-dev/hdmf/blob/dev/src/hdmf/spec/write.py#L28